test/RpcCompatibility.js

```diff
@@ -240,19 +240,31 @@ describe('RpcCompatibility', () => {
         expect(code).to.equal(TestRpcCompatibility.deployedBytecode);
     });
 
-    it('should support RPC method eth_estimateGas', async () => {
-        const newValue = 300;
-        const txParams = {
-            to: rpcCompatibilityContract.address,
-            data: rpcCompatibilityContract.interface.encodeFunctionData('updateValue', [newValue]),
-        };
-
-        const estimatedGas = await provider.send('eth_estimateGas', [txParams]);
-        const gas = BigNumber.from(estimatedGas);
-
-        expect(estimatedGas).to.be.a('string');
-        expect(gas).to.be.gt(0);
-        expect(gas).to.be.lt(30000); // report if gas estimation is different than ethereum
+    describe('eth_estimateGas', () => {
+        it('should support RPC method eth_estimateGas like ethereum mainnet', async () => {
+            const newValue = 300;
+            const txParams = {
+                to: rpcCompatibilityContract.address,
+                data: rpcCompatibilityContract.interface.encodeFunctionData('updateValue', [newValue]),
+            };
+
+            const estimatedGas = await provider.send('eth_estimateGas', [txParams]);
+            const gas = BigNumber.from(estimatedGas);
+
+            expect(estimatedGas).to.be.a('string');
+            expect(gas).to.be.gt(0);
+            expect(gas).to.be.lt(30000); // report if gas estimation does not matches Ethereum's behavior to adjust core configuration if necessary.
+        });
+
+        it('should send tx with estimated gas', async () => {
+            const newValue = 300;
+            const tx = await signer.populateTransaction(await rpcCompatibilityContract.populateTransaction.updateValue(newValue));
```

Comment:
```suggestion
            const tx = await rpcCompatibilityContract.connect(signer).populateTransaction.updateValue(newValue);
```
does this populate all tx fields instead?

Reply:
Yes, you can see the diff here:

```sh
tx 1 {
  data: '0x573c0bd3000000000000000000000000000000000000000000000000000000000000012c',
  to: '0x5FbDB2315678afecb367f032d93F642f64180aa3',
  from: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
  gasLimit: BigNumber { value: "29021528" },
  type: 2,
  maxFeePerGas: BigNumber { value: "3250000000" },
  maxPriorityFeePerGas: BigNumber { value: "1500000000" },
  nonce: 1,
  chainId: 31337
}
tx2 {
  data: '0x573c0bd3000000000000000000000000000000000000000000000000000000000000012c',
  to: '0x5FbDB2315678afecb367f032d93F642f64180aa3',
  from: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
  gasLimit: BigNumber { value: "29021528" }
}
```